sampleExtension/background.js

```diff
@@ -1,60 +1,117 @@
-    updateBlockedSites(replaceListener);
-    addBrowserActionListener();
+/* --------------- ------ update list of BlockedSites ------ ---------------*/
+
+updateStorage = function() {
+    setStorageBlacklist(blockedSites);
+    return {redirectUrl: "https://zeeguu.herokuapp.com/get-ex"};
 };
 
 // This function receives the blacklist from the sync storage.
-updateBlockedSites = function(callback){
-    chrome.storage.sync.get("tds_blacklist", function (items) {
-        if (!chrome.runtime.error) {
-            blockedSites = items.tds_blacklist;
-            return callback();
-        }
+retrieveBlockedSites = function(callback){
+    getStorageBlacklist(function(blacklist) {
+        blockedSites = blacklist;
+        return callback();
+    });
+};
+
+//Loads the intercept time+date list from storage
+retrieveInterceptDateList = function() {
+    getInterceptDateList(function(dateList) {
+        interceptDateList = dateList;
+    });
+};
+
+//Loads the intercept counter from storage
+retrieveInterceptCounter = function(callback) {
+    getInterceptCounter(function(counter) {
+        interceptCounter = counter;
+        return callback();
     });
 };
 
 // This function adds one url to the blacklist
 addToBlockedSites = function(urlToAdd) {
-    var formatUrl = urlToAdd.split(['//'])[1];
-    formatUrl = '*://' + formatUrl + '*';
-    blockedSites.push(formatUrl);
-    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-    });
+    var blockedSiteItem = new BlockedSite(formatUrl(urlToAdd));
+    blockedSites.push(blockedSiteItem);
+    updateStorage();
+    replaceListener();
+};
+
+// This function adds the current time+date to the saved time+date list
+addToInterceptDateList = function() {
+    var newDate = new Date().toDateString();
+    if (interceptDateList == null) {
+        interceptDateList = [newDate];
+    } else {
+        interceptDateList.push(newDate);
+    }
+    setInterceptDateList(interceptDateList);
+};
+
+//TODO fix in interation 3 for url-handling module
+formatUrl = function(url) {
+    result = url.split(['//'])[1];
+    result = result.split("").reverse().join("");
+    result = result.split(['/'])[1];
+    result = result.split("").reverse().join("");
+    return result;
 };
 
 /* --------------- ------ Listener functions ------ ---------------*/
 
 replaceListener = function() {
-    chrome.webRequest.onBeforeRequest.removeListener(intercept);
-    addWebRequestListener();
+    chrome.webRequest.onBeforeRequest.removeListener(handleInterception);
+    urlList = filterBlockedSitesOnChecked();
+    if (urlList.length > 0) {
+        addWebRequestListener(urlList);
+    }
+};
+
+filterBlockedSitesOnChecked = function() {
+    if(blockedSites.length > 0) {
+        return blockedSites.filter(function (a) {return a.checkboxVal == true;});
```

Comment:
The return condition is something that assumes you know the inner composition of the parameter blockedSites. In this javascript file, you declare blockedSites as an empty list and then you use the attribute checkboxVal of the BlockedSite class. What I could point out is that the blockedSites list should be a class in itself, that uses BlockedSite as its list elements. The return condition is part of the BlockedSite class, where you can nicely test for it and then the filter function is part of this new class that handles the list.

This function is itself not that problematic, but my concern is that because you barely use the oop paradigm in your project, this code will soon get out of control.